src/views/Generator/GeneratorTabs/RequestList/RequestList.tsx

```diff
@@ -64,11 +63,6 @@ export function RequestList({
     setShowAllowlistDialog,
   })
 
-  // Preserve the selected request when modifying rules
-  useShallowCompareEffect(() => {
-    onSelectRequest(null)
-  }, [requests])
-
```

Comment:
While it feels like an edge case, the removal of this hook introduced a bug:
1. Open any request for inspection
2. Select a different recording (ideally, with a completely different flow/set of URLs)
3. The inspector will stay open and the request from step 1 will be displayed, while it isn't even present in the new recording

<img width="595" alt="image" src="https://github.com/user-attachments/assets/856ebdb3-4c12-4c7c-b45b-af726c2b7b1d" />